src/components/form/FileInput/FileInput.tsx

```diff
@@ -1,62 +1,130 @@
+     * if set to false input string field will not be presented
+     */
+    showInput?: boolean;
+
+    /**
+     * if set to false reset button will not be presented
+     */
+    showReset?: boolean;
+
+    /**
+     * additional parameters for open file button, props for Button component
+     */
+    openButtonParams?: StrictButtonProps;
+
+    /**
+     * additional help information shown in Popup
+     */
+    help?: string;
+
+    /**
+     * if set to true multiple files can be selected
+     */
+    multiple?: boolean;
+}
+
 /**
  * FileInput is a component showing file input field. Can be displayed as normal input field containing name of the chosen file
  * or as a simple button triggering opening file browser. You can add reset button to unset chosen file.
  *
  * Accessible as `FileInput` or `Form.File`.
  */
-export default function FileInput({
-    value,
-    name,
-    placeholder,
-    onChange,
-    onReset,
-    loading,
-    disabled,
-    showInput,
-    showReset,
-    openButtonParams,
-    help,
-    multiple
-}) {
-    const inputRef = createRef();
+const FileInput: FunctionComponent<FileInputProps> = ({
+    value = null,
+    name = '',
+    placeholder = '',
+    onChange = noop,
+    onReset = noop,
+    loading = false,
+    disabled = false,
+    showInput = true,
+    showReset = true,
+    openButtonParams = {},
+    help = '',
+    multiple = false
+}) => {
+    const inputRef = createRef<HTMLInputElement>();
     const [internalValue, setInternalValue] = useState('');
 
-    const openFileSelection = e => {
-        e.preventDefault();
-        inputRef.current.click();
-        return false;
+    const openFileSelection = (event: React.MouseEvent<HTMLButtonElement>, _data: ButtonProps) => {
+        event.preventDefault();
+        inputRef?.current?.click();
     };
 
     const resetInput = () => {
         setInternalValue('');
         onChange(null, '');
     };
 
-    const resetFileSelection = e => {
-        e.preventDefault();
+    const resetFileSelection = (event: React.MouseEvent<HTMLButtonElement>, _data: ButtonProps) => {
+        event.preventDefault();
         resetInput();
         onReset();
         return false;
     };
 
     const fileChanged = () => {
-        const filename = _.get(inputRef, 'current.files[0].name', '');
-        if (!filename) {
+        if (!inputRef?.current?.files?.[0]) {
             resetInput();
             return;
         }
 
+        const filename = inputRef.current.files[0].name;
         setInternalValue(filename);
         const { files } = inputRef.current;
         if (multiple) onChange(files);
         else onChange(files[0], filename);
+
+        /**
+         * Setting current value to '' to allow uploading the same file again without the need to use reset button
+         * Ref.: https://stackoverflow.com/questions/12030686/html-input-file-selection-event-not-firing-upon-selecting-the-same-file
+         */
+        inputRef.current.value = '';
```

Comment:
WDYT about the refactor below? 🤔 

```suggestion
        const files = inputRef?.current?.files;
        const file = files?.[0];

        if (!file) {
            resetInput();
            return;
        }

        setInternalValue(file.name);
        if (multiple) onChange(files);
        else onChange(file, file.name);

        /**
         * Setting current value to '' to allow uploading the same file again without the need to use reset button
         * Ref.: https://stackoverflow.com/questions/12030686/html-input-file-selection-event-not-firing-upon-selecting-the-same-file
         */
        inputRef.current!.value = '';
```

It decreases the amount of references related to the `inputRef.current.files` and simplifies the view 🎉

Reply:
Good idea, will use it.